ansible/roles/demo_certs/tasks/create-ca.yml

```diff
@@ -3,46 +3,49 @@
 #
 
 ---
-- name: create-ca - Check that Root CA directory {{ root_ca_dir }} exists
+- name: Create-ca - check existence of Root CA directory {{ root_ca_dir }}
   delegate_to: localhost
-  file:
+  ansible.builtin.file:
     path: "{{ root_ca_dir }}"
     state: directory
+    mode: "0600"
   tags:
     - create_ca
 
-- name: create-ca - Copy the ca.conf file to CA
+- name: Create-ca - copy the ca.conf file to CA
   delegate_to: localhost
-  template:
+  ansible.builtin.template:
     src: ca.conf.tpl
     dest: "{{ root_ca_dir }}/ca.conf"
+    mode: "0644"
   tags:
     - create_ca
 
-- name: create-ca - Generate an OpenSSH keypair for the CA
+- name: Create-ca - generate an OpenSSH keypair for the CA
   delegate_to: localhost
-  command:
+  ansible.builtin.command:
     chdir: "{{ root_ca_dir }}"
     cmd: openssl genrsa -out {{ root_ca_dir }}/ca.key 2048
     creates: "{{ root_ca_dir }}/ca.key"
   tags:
     - create_ca
 
-- name: create-ca - Generate self-signed CA certificate
+- name: Create-ca - generate self-signed CA certificate
   delegate_to: localhost
-  command:
+  ansible.builtin.command:
     chdir: "{{ root_ca_dir }}"
     cmd: openssl req -new -x509 -config ca.conf -key ca.key -out ca.crt -days 365 -batch
     creates: "{{ root_ca_dir }}/ca.crt"
   tags:
     - create_ca
 
-- name: create-ca - Initialize CA data files
+- name: Create-ca - initialize CA data files
   delegate_to: localhost
-  shell: |
+  ansible.builtin.shell: |
     cd {{ root_ca_dir }}
     rm -f index.txt serial.txt
     touch index.txt
     echo '01' > serial.txt
+  changed_when: false
```

Comment:
is the changed_when something the linter complained about? seems out of place.

Reply:
yes, and i strongly considered just turning that linter message off